lib/image-diff.js

```diff
@@ -81,8 +81,12 @@ ImageDiff.createDiff = function (options, cb) {
     }
 
     // Callback with pass/fail
-    var totalDifference = resultInfo[1];
-    return cb(null, totalDifference === '0');
+    var percentDifference = parseFloat(resultInfo[2], 10);
+    var threshold = (typeof(options.threshold) === 'undefined' || isNaN(parseFloat(options.threshold, 10))) ? 0.0 : parseFloat(options.threshold);
```

Comment:
There is a lot going on in this line.

- Using `typeof` to detect undefined is only necessary when looking at a global (e.g. `typeof window === 'undefined')
    - Otherwise, we can use a simpler direct comparison with `undefined` (e.g. `options.foo === undefined`)
- We don't need to detect `undefined`, a fallback will work fine (e.g. `var foo = options.bar || baz;`)
- We don't need to be defensive and coerce `options.threshold`. We can assume the developer wants the library to work and will pass in a `Number` (i.e. no `isNaN/parseFloat` necessary)
- In JS, all `Numbers` are 64 bit floats. The decimal isn't necessary; `0` will work fine.
    - https://developer.mozilla.org/en-US/docs/Glossary/Number

The simplified result should be:

```js
var threshold = options.threshold || 0;
```